### PR TITLE
Android: Serial Fixes

### PR DESCRIPTION
**Pull request by HTRamsey**
- Remove custom SerialInputOutputManager and use default from library 
- Improve QSerialPortInfo data (was missing a couple of values)
- Create QGCSerialListener class
- Fix some miscellaneous issues and remove some unused functions

### CODE REVIEW

File: android/libs/qtandroidserialport/qserialport_android.cpp [hunk 1]
```diff
@@ -38,12 +38,12 @@ bool QSerialPortPrivate::open(QIODevice::OpenMode mode)
         return false;
     }
 
-    /*if (mode & QIODevice::ReadOnly) {
```
Comment (DonLakeFlyer) [inline android/libs/qtandroidserialport/qserialport_android.cpp:41]: What's with this uncommented and then commented out stuff?

File: android/AndroidManifest.xml [hunk 2]
```diff
@@ -56,7 +58,6 @@
             <intent-filter>
                 <action android:name="android.hardware.usb.action.USB_DEVICE_ATTACHED" />
                 <action android:name="android.hardware.usb.action.USB_DEVICE_DETACHED" />
-                <action android:name="android.hardware.usb.action.USB_ACCESSORY_ATTACHED" />
```
Comment (DonLakeFlyer) [inline android/AndroidManifest.xml:59]: What's the distinction between USB_DEVICE and USB_ACCESORY?

Comment (HTRamsey) [reply android/AndroidManifest.xml:59]: Accessory was used for Taisync before that was removed. It's just which end acts as the USB Host. https://developer.android.com/develop/connectivity/usb

File: android/libs/qtandroidserialport/qserialport_android.cpp (hunk 1 above)
Comment (HTRamsey) [reply android/libs/qtandroidserialport/qserialport_android.cpp:41]: I tried to model this file after Qt's Unix class, but some commented out code doesn't work as I expected yet or I'm not sure it's necessary. I'll either fix it or remove it in a follow up after more testing.

**Pull request merged**